website/src/components/Tasks/Task.tsx

```diff
@@ -18,51 +37,85 @@ export const Task = ({ tasks, trigger, mutate }) => {
     },
   });
 
-  const rejectTask = (task: { id: string }, reason: string) => {
+  const rejectTask = (reason: string) => {
     sendRejection({
       id: task.id,
       reason,
     });
   };
 
-  function taskTypeComponent(type) {
-    const taskType = TaskTypes.find((taskType) => taskType.type === type);
-    const category = taskType.category;
-    switch (category) {
+  const onReplyChanged = useRef((state: TaskReplyState<TaskContent>) => {
+    if (taskStatus === "SUBMITTED") return;
+
+    replyContent.current = state?.content;
+    if (state === null) {
+      if (taskStatus !== "NOT_SUBMITTABLE") setTaskStatus("NOT_SUBMITTABLE");
+    } else if (state.state === "DEFAULT") {
+      if (taskStatus !== "DEFAULT") setTaskStatus("DEFAULT");
+    } else if (state.state === "VALID") {
+      if (taskStatus !== "SUBMITABLE") setTaskStatus("SUBMITABLE");
+    }
+  }).current;
+
+  const submitResponse = () => {
+    switch (taskStatus) {
+      case "NOT_SUBMITTABLE":
+        return;
+      case "DEFAULT":
+        setShowUnchangedWarning(true);
+        break;
+      case "SUBMITABLE": {
+        trigger({
+          id: task.id,
```

Comment:
This breaks things.  task.id isn't the frontend ID it's the backend ID.  Right now we can't do a guaranteed lookup to find the proper backend ID. 

Once the backend changes remove the frontend ID this will be correct but as of right now submissions are broken.

Reply:
Ah, right. I can look into fixing it in about 5hrs, or do you want to fix it before then?

Reply:
I've got my fix up and ready :)